enumerable.rb

```diff
@@ -0,0 +1,89 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    size.times { |n| yield (self[n]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    size.times { |n| yield self[n], n }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result_array = []
+    my_each { |n| result_array << n if yield(n) }
+    result_array
+  end
+
+  def my_all
+    return true unless block_given?
+
+    my_each { |n| return false unless yield(n) }
+    true
+  end
+
+  def my_any
+    return true unless block_given?
+
+    my_each { |n| return true if yield(n) }
+    false
+  end
+
+  def my_none
+    return true unless block_given?
+
+    my_each { |i| return false if yield(i) == true }
+    true
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+
+    if !arg.nil?
+      my_each { |i| counter += 1 if i == arg }
+    elsif block_given?
+      my_each { |i| counter += 1 if yield(i) }
+    else
+      counter
+    end
+    counter
+  end
+
+  def my_map(a_proc = nil)
+    return to_enum unless block_given?
```

Comment:
### **my_map**

- [ ] executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum

Reply:
Ruby won't let us supply both a proc and block at the same time, so I am not sure how we are supposed to do this if it cant run in the first place when both are supplied
![error](https://user-images.githubusercontent.com/51087315/83785011-584efc80-a699-11ea-8809-94fc69effe75.jpg)